static/app/components/core/badge/featureBadge.tsx

```diff
@@ -63,44 +45,81 @@ const useIndicatorColor = () => {
     new: theme.green300,
     experimental: theme.gray100,
     internal: theme.gray100,
-  } satisfies Record<FeatureBadgeType, string>;
+  } satisfies Record<FeatureBadgeProps['type'], string>;
 };
 
-function BaseFeatureBadge({
+export interface FeatureBadgeProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'title'> {
+  type: 'alpha' | 'beta' | 'new' | 'experimental' | 'internal';
+  expiresAt?: Date;
+  tooltipProps?: Partial<TooltipProps>;
+  variant?: 'badge' | 'indicator' | 'short';
+}
+
+function InnerFeatureBadge({
   type,
   variant = 'badge',
-  title,
   tooltipProps,
   expiresAt,
   ...props
-}: Props) {
-  const indicatorColors = useIndicatorColor();
+}: FeatureBadgeProps) {
+  const indicatorColors = useFeatureBadgeIndicatorColor();
+
+  if (
+    hasFeatureBadgeExpired(
+      expiresAt,
+      tooltipProps?.title?.toString() ?? '<unknown title>',
+      type
+    )
+  ) {
+    return null;
+  }
+
+  const {title, ...restTooltipProps} = tooltipProps ?? {};
+
+  return (
+    <div {...props}>
+      <Tooltip
+        title={title ?? defaultTitles[type]}
+        position="right"
+        {...restTooltipProps}
+      >
+        {variant === 'badge' || variant === 'short' ? (
+          <StyledBadge
+            type={type}
+            text={variant === 'short' ? shortLabels[type] : labels[type]}
+          />
+        ) : (
+          <CircleIndicator color={indicatorColors[type]} size={8} />
+        )}
+      </Tooltip>
+    </div>
+  );
+}
+
+/**
+ * Checks if a feature badge has expired - if it has, reports the result to Sentry
+ * @param expiresAt The date the feature badge expires.
+ * @returns True if the feature badge has expired, false otherwise.
+ */
+function hasFeatureBadgeExpired(
+  expiresAt: Date | undefined,
+  title: string,
+  type: FeatureBadgeProps['type']
+) {
   if (expiresAt && expiresAt.valueOf() < Date.now()) {
     // Only get 1% of events as we don't need many to know that a badge needs to be cleaned up.
     if (Math.random() < 0.01) {
       withScope(scope => {
-        scope.setTag('title', title?.toString());
+        scope.setTag('title', title);
         scope.setTag('type', type);
         scope.setLevel('warning' as SeverityLevel);
         captureException(new Error('Expired Feature Badge'));
       });
     }
-    return null;
+    return true;
   }
-
-  return (
-    <div {...props}>
-      <Tooltip title={title ?? defaultTitles[type]} position="right" {...tooltipProps}>
```

Comment:
This is actually hiding a bug - tooltipProps has a title member, meaning that if we were to use `<FeatureBadge title='something' tooltipProps={{title: 'something else'}} />`, the final tooltip output would only contain 'something else'